File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    num = sorted(list(map(int, input().split())))

    # 0 0 0 입력이 들어오면 반복 종료
    if num[0] == num[1] == num[2] == 0:
        break

    # 삼각형이 유효하지 않은 경우 (두 짧은 변의 합이 가장 긴 변보다 작거나 같으면 삼각형이 될 수 없음)
    if num[0] + num[1] <= num[2]:
        print("Invalid")
    # 세 변의 길이가 모두 같은 경우
    elif num[0] == num[1] == num[2]:
        print("Equilateral")
    # 두 변의 길이가 같은 경우
    elif num[0] == num[1] or num[1] == num[2] or num[2]==num[0]:
        print("Isosceles")
    # 세 변의 길이가 모두 다른 경우
    else:
        print("Scalene")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;

import java.util.Arrays;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		while (true) {
			String[] xyz = br.readLine().split(" ");
			int[] triangle_sides = str_to_int(xyz);
			
			if (isValid_angle(triangle_sides)) {
				String res = my_detecter(triangle_sides);
				System.out.println(res);
			}
			else if (!isValid_angle(triangle_sides)) {
				break;
			}
		}
		
	}
	
	
	
	public static String my_detecter(int[] int_arr) {
		String res = "";
		
		int x = int_arr[0];
		int y = int_arr[1];
		int z = int_arr[2];
		
		
		if (x == y && y == z && z == x) {
			res = "Equilateral";
		}
		else if (x + y <= z) {
			res = "Invalid";
		}
		else if ( (x==y && y != z) ||  (y == z && z != x) || (z == x && x != y) ) {
			res = "Isosceles";
		}
		else if (x != y && y != z && z != x) {
			res = "Scalene";
		}
		
		return res;
	}
	
	
	
	
	
	public static boolean isValid_angle(int[] int_arr) {
		int cnt = 0;
		for (int i = 0; i < int_arr.length; i++) {
			if (int_arr[i] == 0) cnt++;
		}
		if (cnt == 3) {
			return false;
		}
		return true;
	}
	
	
	
	public static int[] str_to_int(String[] arr) {
		int[] res = new int[arr.length];
		for (int i = 0; i < arr.length; i++) {
			res[i] = Integer.parseInt(arr[i]);
		}
		Arrays.sort(res);
		return res;
	}
	
}